chromium_src/chrome/browser/history/history_utils.cc

```diff
@@ -7,9 +7,22 @@
 #include "../../../../../chrome/browser/history/history_utils.cc"  // NOLINT
 #undef CanAddURLToHistory
 
+#if defined(ENABLE_EXTENSION)
+#include "brave/common/extensions/extension_constants.h"
+#endif
+
 bool CanAddURLToHistory(const GURL& url) {
   if (!CanAddURLToHistory_ChromiumImpl(url))
     return false;
 
-  return !url.SchemeIs(content::kBraveUIScheme);
+  bool is_brave_scheme = url.SchemeIs(content::kBraveUIScheme);
+
+#if !defined(ENABLE_EXTENSION)
+  return !is_brave_scheme;
+#else
+  bool is_wallet_host =
```

Comment:
please add a `url.scheme()` check here for extensions as well

Reply:
done